doc: add documentation on how to handle unavailable tapes

YoannV-Ayudata opened the pull request:
This patch simply adds the documentation to specify the procedure explained in https://github.com/phobos-storage/phobos/issues/31. If validated, the PR will be closed without being merged, the instead the document will be added to the wiki of the main repository.

valeriyoann committed:
doc: add documentation on how to handle unavailable tapes

SebaGougeaud commented on doc/handling_unavailable_tapes.md:
```diff
@@ -0,0 +1,176 @@
+
+## Phobos side
+
+On Phobos's side, there are 4 steps to declaring a tape lost, ensuring Phobos
```
typo: `to declare [...], ensure [...] and handle [...]`

SebaGougeaud commented on doc/handling_unavailable_tapes.md:
```diff
@@ -0,0 +1,176 @@
+## Phobos side
+
+On Phobos's side, there are 4 steps to declaring a tape lost, ensuring Phobos
+won't use it again, and handling the objects that may not be available anymore.
```
typo: `will not`, no abbreviated form in writing

SebaGougeaud commented on doc/handling_unavailable_tapes.md:
````diff
@@ -0,0 +1,176 @@
+phobos tape lock <tape label>
+```
+
+With this, if a client tries to get an object or copy whose's content is only
````
typo: `whose`

SebaGougeaud commented on doc/handling_unavailable_tapes.md:
````diff
@@ -0,0 +1,176 @@
+thus all the data stored on it cannot be accessed anymore. At this point, the
+first thing to do is to prevent Phobos from trying to use that tape, which is
+done by using the `lock` command as such:
+```
+phobos tape lock <tape label>
+```
````
style: I think markdown prefer when there is a blank line before and after this code block

SebaGougeaud commented on doc/handling_unavailable_tapes.md:
```diff
@@ -0,0 +1,176 @@
+ - For RAID1 layouts, if the copy has a replica count superior to 1 and the
+other extents for that copy are on available media, the object is still
+readable.
+ - For RAID4 layouts, if the two other extents for that copy has readable, then
```
typo: `are readable`? maybe `available`

SebaGougeaud commented on doc/handling_unavailable_tapes.md:
```diff
@@ -0,0 +1,176 @@
+Note: when deleting objects with extents on tape, Phobos will *not* actually
+delete the extents from the tape or from the database. For the latter, Phobos
+will simply change the extents' state to `orphan` (i.e. not related to any copy
+or object) in the database, and expect a later call to the garbage collector to
```
comment: I think it is not relevant to talk of the garbage collector as we do not have one yet

SebaGougeaud commented on doc/handling_unavailable_tapes.md:
```diff
@@ -0,0 +1,176 @@
+delete the extents from the tape or from the database. For the latter, Phobos
+will simply change the extents' state to `orphan` (i.e. not related to any copy
+or object) in the database, and expect a later call to the garbage collector to
+delete the database's entry. For the former, the data is expected to be deleted
```
typo: `database entries`?

SebaGougeaud commented on doc/handling_unavailable_tapes.md:
```diff
@@ -0,0 +1,176 @@
+
+### Fourth step (optional)
+
+The last step on Phobos' side is optionnal, and is to remove the extents from
```
typo: at the start of this document, it is written `Phobos's side`, now `Phobos' side`, we need to stick to one spelling

SebaGougeaud commented on doc/handling_unavailable_tapes.md:
```diff
@@ -0,0 +1,176 @@
+If the command suceeds, then the tape will be inserted back into the system as
+a fresh new one, and Phobos may use it in the future to try and hold data.
+
+For the fourth state, ythe tape can be inserted back into Phobos as an already
```
typo: `ythe` -> `the`

SebaGougeaud commented on doc/handling_unavailable_tapes.md:
````diff
@@ -0,0 +1,176 @@
+phobos tape import <tape label>
+```
+
+This command will add the tape into the system, and read its content to register
+any extent found. When done reading the content, it will associate any extent
+found with their copies and objects, which may mean that objects impacted by
+the loss of the tape will be readable/whole again.
````
question: do we need to ensure this is doable by phobos by implementing a test?

SebaGougeaud commented on doc/handling_unavailable_tapes.md:
```diff
@@ -0,0 +1,176 @@
+
+## Future features
+
+We currently have 4 features planned related to this procedure to make it
+easier:
+ - the `extent delete` command to delete an extent from the database and update
+the copy and object linked to it, while also showing if the copy and object are
+still readable or not (it is planned for version 3.2)
+ - the `extent delete --orphan` command to automatically delete extents flagged
+as `orphan` in the database (it is planned for version 3.2)
+ - the `tape delete --lost` command to delete a tape from the database and
+handle the impacted extents, copies and objects accordingly
+ - the `tape export` command to remove a tape from the system but handling it
+as if it was to come back one day.
```
question: don't you want to also talk of the garbage collector here?

SebaGougeaud reviewed (changes requested):
Ok with the general talk. Not a Lustre guru, so cannot ensure the Lustre part is correct. I will approve the PR once the question about having a test for this scenario is answered.

valeriyoann committed:
doc: fix documentation issues in handle_unavailable_tapes.md

YoannV-Ayudata committed:
doc: add missing `--name` is documentation command

YoannV-Ayudata commented on doc/handling_unavailable_tapes.md:
```diff
@@ -0,0 +1,184 @@
+
+We currently have 4 features planned related to this procedure to make it
+easier:
+ - the `extent delete` command to delete an extent from the database and update
+the copy and object linked to it, while also showing if the copy and object are
+still readable or not (it is planned for version 3.2)
+ - the `extent delete --orphan` command to automatically delete extents flagged
+as `orphan` in the database (it is planned for version 3.2)
```
Removed the item and let the garbage collector handle this. Haven't added a command example though, as we don't even know the expected syntax yet (we don't know about `extent delete` either, but it is way more likely to be this one).

YoannV-Ayudata committed:
doc: remove example of `extent delete --orphan`

Because it is a duplicate of the GC policy and inconsistent with the
expected `extent delete` command.

Change-Id: Iaf74cea6576ddb7452afc9a240069c9646f60836
Signed-off-by: VALERI Yoann <yoann.valeri.externe@ayudata.fr>

YoannV-Ayudata committed:
Merge branch 'master' into doc

Change-Id: Ib29907eca507e27f7e677dfb587eca47bec28207
Signed-off-by: VALERI Yoann <yoann.valeri.externe@ayudata.fr>

YoannV-Ayudata committed:
doc: update doc + print new copy status on delete

Change-Id: I63299c1c18c66732a0d2d2aeb1c829c417494011
Signed-off-by: VALERI Yoann <yoann.valeri.externe@ayudata.fr>
